File: db4oj.tests/src/com/db4o/db4ounit/common/io/RandomAccessFileStorageFactoryTestCase.java
/* Copyright (C) 2009  Versant Inc.  http://www.db4o.com */
package com.db4o.db4ounit.common.io;

import com.db4o.db4ounit.common.api.*;
import com.db4o.io.*;

import db4ounit.*;
import db4ounit.extensions.*;

public class RandomAccessFileStorageFactoryTestCase extends TestWithTempFile {
	
	private final Storage subject = Db4oUnitPlatform.newPersistentStorage();
	
	public void testExistsWithUnexistentFile() {
		Assert.isFalse(subject.exists(tempFile()));
	}
	
	public void testExistsWithZeroLengthFile() {
		final Bin storage = subject.open(new BinConfiguration(tempFile(), false, 0, false));
		storage.close();
		Assert.isFalse(subject.exists(tempFile()));
	}	

}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/IO/RandomAccessFileStorageFactoryTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4oUnit.Extensions;
using Db4objects.Db4o.IO;
using Db4objects.Db4o.Tests.Common.Api;

namespace Db4objects.Db4o.Tests.Common.IO
{
	public class RandomAccessFileStorageFactoryTestCase : TestWithTempFile
	{
		private readonly IStorage subject = Db4oUnitPlatform.NewPersistentStorage();

		public virtual void TestExistsWithUnexistentFile()
		{
			Assert.IsFalse(subject.Exists(TempFile()));
		}

		public virtual void TestExistsWithZeroLengthFile()
		{
			IBin storage = subject.Open(new BinConfiguration(TempFile(), false, 0, false));
			storage.Close();
			Assert.IsFalse(subject.Exists(TempFile()));
		}
	}
}
